apple/testing/default_runner/simulator_creator.py

```diff
@@ -66,39 +68,54 @@ def _build_parser() -> argparse.ArgumentParser:
         default=None,
         help="The name to use for the device; default is 'BAZEL_TEST_[device_type]_[os_version]'",
     )
+    parser.add_argument(
+        "--reuse-simulator",
+        action=argparse.BooleanOptionalAction,
+        default=True,
+        help="Toggle simulator reuse; default is True",
+    )
     return parser
 
 
-def _main(os_version: str, device_type: str, name: Optional[str]) -> None:
+def _main(os_version: str, device_type: str, name: Optional[str], reuse_simulator: bool) -> None:
     devices = json.loads(_simctl(["list", "devices", "-j"]))["devices"]
     device_name = name or _device_name(device_type, os_version)
     runtime_identifier = "com.apple.CoreSimulator.SimRuntime.iOS-{}".format(
         os_version.replace(".", "-")
     )
 
-    devices_for_os = devices.get(runtime_identifier) or []
-    existing_device = next(
-        (blob for blob in devices_for_os if blob["name"] == device_name), None
-    )
+    existing_device=None
+
+    if reuse_simulator:
+        devices_for_os = devices.get(runtime_identifier) or []
+        existing_device = next(
+            (blob for blob in devices_for_os if blob["name"] == device_name), None
+        )
 
     if existing_device:
         simulator_id = existing_device["udid"]
+        name = existing_device["name"]
         # If the device is already booted assume that it was created with this
         # script and bootstatus has already waited for it to be in a good state
         # once
         state = existing_device["state"].lower()
-        print(f"Simulator state is: {state}", file=sys.stderr)
+        print(f"Existing simulator '{name}' ({simulator_id}) state is: {state}", file=sys.stderr)
         if state != "booted":
             _boot_simulator(simulator_id)
     else:
+        # Simulator reuse is based on device name, therefore we must generate a unique name to
+        # prevent unintended reuse.
+        device_name_suffix = ''.join(random.choices(string.ascii_letters + string.digits, k=8))
```

Comment:
doesn't  this mean the first time we're reusing but there isn't an existing device we'll create one with a random name like this?

Reply:
Good call, I just pushed a fix for this.

```
$ python3 simulator_creator.py "16.2" "iPhone 8 Plus"
Created new simulator 'BAZEL_TEST_iPhone 8 Plus_16.2' (9B2AE56C-88C5-49F6-9E06-ED3E5115EE5A)
Monitoring boot status for BAZEL_TEST_iPhone 8 Plus_16.2 (9B2AE56C-88C5-49F6-9E06-ED3E5115EE5A).
Booting device...
...
9B2AE56C-88C5-49F6-9E06-ED3E5115EE5A

$ python3 simulator_creator.py "16.2" "iPhone 8 Plus"
Existing simulator 'BAZEL_TEST_iPhone 8 Plus_16.2' (9B2AE56C-88C5-49F6-9E06-ED3E5115EE5A) state is: booted
9B2AE56C-88C5-49F6-9E06-ED3E5115EE5A

$ python3 simulator_creator.py "16.2" "iPhone 8 Plus" --no-reuse-simulator
Created new simulator 'BAZEL_TEST_iPhone 8 Plus_16.2_1O1R6VdK' (3599B1D0-7A13-4ECA-917C-72F7B85E37BE)
Monitoring boot status for BAZEL_TEST_iPhone 8 Plus_16.2_1O1R6VdK (3599B1D0-7A13-4ECA-917C-72F7B85E37BE).
Booting device...
...
3599B1D0-7A13-4ECA-917C-72F7B85E37BE

$ python3 simulator_creator.py "16.2" "iPhone 8 Plus"
Existing simulator 'BAZEL_TEST_iPhone 8 Plus_16.2' (9B2AE56C-88C5-49F6-9E06-ED3E5115EE5A) state is: booted
9B2AE56C-88C5-49F6-9E06-ED3E5115EE5A

$ python3 simulator_creator.py "16.2" "iPhone 8 Plus" --no-reuse-simulator
Created new simulator 'BAZEL_TEST_iPhone 8 Plus_16.2_Hjqetj1T' (6A1EE2BA-519E-411C-8653-5F6C375E989D)
Monitoring boot status for BAZEL_TEST_iPhone 8 Plus_16.2_Hjqetj1T (6A1EE2BA-519E-411C-8653-5F6C375E989D).
Booting device...
...
6A1EE2BA-519E-411C-8653-5F6C375E989D
```